src/AutoDiscover.php

```diff
@@ -625,11 +625,19 @@ public function toXml()
     }
 
     /**
-     * Handle WSDL document.
+     * Handle WSDL document
+     *
+     * @param  string|null|callable $responseHandler
+     * @return void
      */
-    public function handle()
+    public function handle($responseHandler = null)
```

Comment:
@alexmerlin 
The class is not final therefore this cannot be correct.

Reply:
@froschdesign We have made the class final. Please review once.

Reply:
@Atul-glo35265 make class final is bc break, if it required, uses `func_get_args()` may can handle that, see https://3v4l.org/8na14

Reply:
@Atul-glo35265 
Changing the method signature and making a class final represent a break in backwards compatibility.

The question that arises here is why was the method changed at all? Only for tests?

Reply:
@froschdesign There were 3 errors in [QA Checks (PHPUnit [8.4, lowest] failure](https://github.com/laminas/laminas-soap/actions/runs/11327465531/job/31498597871). We made changes in these respective methods to fix the issue.

I'll try to implement if we can achieve the same result by implementing func_get_args() as suggested by [samsonasik](https://github.com/samsonasik)

Reply:
@Atul-glo35265 
> We made changes in these respective methods to fix the issue.

The error is not in the `AutoDiscover` class and `func_get_args` will not help here, as the logic of the method does not require an external parameter.
Check and correct the tests and not the implementation.

Reply:
Yes @froschdesign. I'm trying the same on the local machine.

Reply:
Change the version in the Composer configuration file. See:

- https://github.com/sebastianbergmann/phpunit/releases/tag/9.6.21
- https://github.com/sebastianbergmann/phpunit/releases/tag/9.6.18

Reply:
@froschdesign I've updated the version to 9.6.21 but the issue is still there.

Actually, the annotation **@runinseparateprocess** is causing the issue and throwing the exception. If we remove this annotation, issues related to _Header already sent_ start coming.

I tried to upgrade PHPUnit to version **^10.0.0**, but it seems we need to upgrade other dependencies as well.

Reply:
@Atul-glo35265 
> I've updated the version to 9.6.21 but the issue is still there.

The version of PHPUnit must be changed in the `composer.json` file!

Reply:
I have updated it in composer.json file. Please check and suggest if any other changes are required.

Reply:
Bump also all other packages in `require-dev` section.

https://getcomposer.org/doc/03-cli.md#bump

Reply:
If `bump`ing does not work, you can upgrade them manually. I just tested, they can be upgraded.